tests/opwen_email_server/integration/test_celery.py

```diff
@@ -3,38 +3,80 @@
 from string import ascii_letters
 from unittest import TestCase
 from unittest import skipUnless
+import ast
+import inspect
 
 from cached_property import cached_property
 from kombu import Connection
 from kombu import Exchange
 from kombu import Queue
 
 from opwen_email_server.config import QUEUE_BROKER
+import opwen_email_server.integration.celery as celery
 
 
 class TransportTests(TestCase):
-    exchange_name = getenv('KOMBU_EXCHANGE', 'testexchange')
-    queue_name = getenv('KOMBU_QUEUE', 'testqueue')
-    routing_key = getenv('KOMBU_ROUTING_KEY', 'testkey')
-
-    @cached_property
-    def exchange(self) -> Exchange:
-        return Exchange(self.exchange_name, 'direct', durable=True)
-
-    @cached_property
-    def queue(self) -> Queue:
-        return Queue(self.queue_name, exchange=self.exchange, routing_key=self.routing_key)
-
-    @skipUnless(QUEUE_BROKER, 'no celery broker configured')
-    def test_send_message(self):
-        random_message = ''.join(choice(ascii_letters)  # nosec
-                                 for _ in range(30))
-
-        with Connection(QUEUE_BROKER) as connection:
-            producer = connection.Producer()
-            producer.publish(
-                {'message': random_message, 'test': True},
-                exchange=self.exchange,
-                routing_key=self.routing_key,
-                declare=[self.queue],
-            )
+	exchange_name = getenv('KOMBU_EXCHANGE', 'testexchange')
+	queue_name = getenv('KOMBU_QUEUE', 'testqueue')
+	routing_key = getenv('KOMBU_ROUTING_KEY', 'testkey')
+
+	@cached_property
+	def exchange(self) -> Exchange:
+		return Exchange(self.exchange_name, 'direct', durable=True)
+
+	@cached_property
+	def queue(self) -> Queue:
+		return Queue(self.queue_name, exchange=self.exchange, routing_key=self.routing_key)
+
+	@skipUnless(QUEUE_BROKER, 'no celery broker configured')
+	def test_send_message(self):
+		random_message = ''.join(choice(ascii_letters)  # nosec
+								 for _ in range(30))
+
+		with Connection(QUEUE_BROKER) as connection:
+			producer = connection.Producer()
+			producer.publish(
+				{'message': random_message, 'test': True},
+				exchange=self.exchange,
+				routing_key=self.routing_key,
+				declare=[self.queue],
+			)
+	
+	def test_celery_task_routes(self):
+		# REFERENCE: https://stackoverflow.com/questions/3232024/introspection-to-get-decorator-names-on-a-method
```

Comment:
This seems like a fairly complex approach. Would it be possible to simplify this via [the Celery task registry](https://docs.celeryproject.org/en/latest/reference/celery.html#celery.Celery.tasks) ([see StackOverflow](https://stackoverflow.com/q/26058156/3817588)) and filtering out the system tasks which all seem to be prefixed with `celery.`?

```py
from opwen_email_server.integration.celery import celery as app  # ensure app tasks are loaded

print([task_name for task_name in app.tasks.keys() if not task_name.startswith('celery.')])
```